x-pack/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/index.tsx

```diff
@@ -28,6 +27,12 @@ import { MaybeViewTraceLink } from './MaybeViewTraceLink';
 import { TransactionTabs } from './TransactionTabs';
 import { IWaterfall } from './WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers';
 
+type DistributionApiResponse = APIReturnType<
+  'GET /api/apm/services/{serviceName}/transaction_groups/distribution'
+>;
+
+type DistributionBucket = DistributionApiResponse['buckets'][0];
```

Comment:
I usually use `ValuesType` (from `utility-types`), as it works for both arrays and objects, and uses a `number` as an indexer. If you have a tuple for instance, if you use 0 as an indexer, the type would be of the first item of the array, but with ValuesType it would (more correctly) be the union type of all the items in the array. But it's more typing ofc, and `[0]` is correct in most of the cases. 😀

Reply:
Cool, I'm good with using `ValueType` instead of `[0]`. I think there is a bunch of other places where we already use `[0]`. You okay with merging this as-is and then we change it everywhere in another PR (or gradually)?